### ISSUE TITLE
Improve "upgrade app" modal

### ISSUE THREAD

**Issue by jsilvestre** [cozy/cozy-home#291]
Based on @goofy-bz suggestion on the forum: https://forum.cozy.io/t/pendant-une-mise-a-jour-fenetre-dinformation-trop-fugace/215/3

The first screen of the modal is a "please wait" message that last the time the permissions are downloaded. It oftens just blinks because it's fast.
* it should not be displayed if it's fast
* it should be displayed enough time so the user can read it if it's not that fast
Either (1) or (2), or a mix of both.

If we display a "waiting" screen, it should not display buttons that look to be clickable as it's misleading for the user that thinks it missed a step.

**Comment by m4dz** [comment]
My tech thoughts about the waiting / throbber things is:
- apply an attribute (ideally an `aria-busy=true`) on the currently waiting element
- using CSS transition, let a pseudo-element containing the throbber appear _after a delay_ (about something like 850ms) on the targeted element.

This way, we instantly mark the element as busy, but inform user only if it take a reasonable amount of time to be completed. This will prevent the user feeling that you must waiting for something but finally not. As it only use CSS to display feedback to the user, we don't have to deal with that on the logic part.

In this case, we must display a message to the user, so I suggest to use an element inside the box with attribute [`role=status`](http://www.w3.org/TR/wai-aria/roles#status) which can contains the message and that we manipulate _via_ CSS like the pseudo-element mentionned before (i.e. `[aria-busy=true] [role=status]{ display: block }`).

Thoughts?

**Comment by frankrousseau** [comment]
+1 with @m4dz

**Comment by jsilvestre** [comment]
I like your solution, but that doesn't entirely solve the case: if it takes 1300ms tor retrieve the data, the throbber message will be displayed only for 250ms, resulting in the throbber blink. Maybe we could use the same transition technique so it disappear after 2s (placeholder value) in any case?

**Comment by m4dz** [comment]
Hum, you're right… 2s may be too long just to hide a notification, but what if we set a transition duration of ~1s for hiding ? It defines the following flow:
1. Need a notification, set `aria-busy` to `true`, it starts transition delayed by 850ms
2.
    1. Busy state < 850ms : `aria-busy` set to `false`, transition cancelled > END
    2. Busy state > 850ms : transition occurs, notification is visible
3. Busy state is remove : `aria-busy` set to `false`, starting transition to hide notification during 1s, and showing block content (that is now ready) with same 1s transition.

If we add a backdrop to the notification, we can show the content under the notification message and fade the both elements without having an unpleasant effect. Even if the busy state is removed when message appear, it will remains on screen for 1s (the fading transition), so it won't blink.

We probably have to adjust times with tests to ensure the effect is smooth.

Is it better ?

**Comment by jsilvestre** [comment]
Yeah I think that's good. About the exact time, I just don't know, experience will tell. It's just about the user being able to read it!
